### ISSUE TITLE
Refs that do not belong to ostree should not be automatically pruned

### ISSUE THREAD

**Issue by jlebon** [ostreedev/ostree#262]
There are multiple apps now that also need to store data in the system repo. These apps might not want ostree to automatically prune their refs (see `ostree_admin_cleanup()`).

We need to make ostree smarter so that it only prunes the refs that belong to it.
See also PR #257.

**Comment by cgwalters** [comment]
So...we ended up using a separate repo for this, but I have a suspicion this is still going to bite people down the line.

**Comment by jlebon** [comment]
Yup, for this reason I had to add refs to the base layers for package layering (although to be fair, in that case the commit *does* "belong" to ostree, but we should probably still have some configurability here so we don't have to hack around it).
